[ENH] Reorganise modality agnostic files section into multiple pages

bclenet opened the pull request:
BEP028 BIDS-Prov is going to introduce a lot of contents into the *Modality agnostic files* section of the BIDS specification.

In order to prepare for these changes, this PR splits the contents of the `src/modality-agnostic-files.md` file into separate files in a `src/modality-agnostic-files/` directory:
 * `src/modality-agnostic-files/dataset-description.md`
 * `src/modality-agnostic-files/data-description.md` 
 * `src/modality-agnostic-files/code.md`

bclenet commented on mkdocs.yml:
```diff
@@ -101,21 +104,15 @@ extra:
 extra_javascript:
   - js/jquery-3.6.0.min.js
-extra_css:
-  - css/tsv.css
```
These changes are indeed not related to the PR. I'm not sure to understand how they ended up here but I'll fix that as soon as possible.

bclenet commented:
Hi !
CI seems fails (`check_links`) fails because of a link that is not related to the PR. Could someone please relaunch the CI pipeline so that this fail disappears ?
Thanks

cmaumet commented:
Hi everyone. Is it okay to merge this PR or are there more updates that would be required? As a note, this is only a refactoring of existing info (no new and/or no remove text). Thanks!

yarikoptic commented on src/CHANGES.md:
```diff
@@ -625,7 +625,7 @@
 
 -   Added T1 Rho maps: [4.1 Anatomy imaging data](modality-specific-files/magnetic-resonance-imaging-data.md#anatomy-imaging-data).
--   Added support for phenotypic information split into multiple files: [3.2 Participant key file](modality-agnostic-files.md#participants-file).
+-   Added support for phenotypic information split into multiple files: 3.2 Participant key file.
```
return reference here?

yarikoptic commented on src/pregh-changes.md:
```diff
@@ -32,7 +32,7 @@
 
 -   Added T1 Rho maps: [4.1 Anatomy imaging data](modality-specific-files/magnetic-resonance-imaging-data.md#anatomy-imaging-data).
--   Added support for phenotypic information split into multiple files: [3.2 Participant key file](modality-agnostic-files.md#participants-file).
+-   Added support for phenotypic information split into multiple files: 3.2 Participant key file.
```
and here is another copy of it seems to me

yarikoptic reviewed (changes requested):
I am a little uncertain about using "Data" here, since "Dataset" includes "Data" as well.
Especially since to a degree summaries like `participants.tsv` are also "dataset level description": although they already were separated from "Dataset description" section, they were not explicitly sectioned under "Data" section. 

here is what is there:
![image](https://github.com/user-attachments/assets/7548bd55-cbea-468c-a0f5-4c3cb56ec4b5)

It seems that what is common among all, but `phenotype/`, those "Data description" files -- they are directly pertinent to data under `sub-*` folders.  `phenotype/`, sure thing, also but it is under a separate folder.  I wonder if we should

- also move out [Phenotypic and assessment data](https://bids-specification.readthedocs.io/en/stable/modality-agnostic-files.html#phenotypic-and-assessment-data) into its own file/sub-section.
- rename "Data description" into "Data summary files" since all those sections are "{entity/concept} files" (`concept` is for `scans` one)
   - this would also better align with a potential bids-2 level enhancement of https://github.com/bids-standard/bids-2-devel/issues/65 and https://github.com/bids-standard/bids-2-devel/issues/85 .

bclenet commented:
Hi @yarikoptic thanks for the review, I changed the names of the  "Data description" section and added a new one for Phenotypic and assessment data, as you suggested.

yarikoptic reviewed (approved):
<details>
<summary>IMHO looks nice and now more consistent with "Modality specific files" section being expandable too</summary> 

![image](https://github.com/user-attachments/assets/5da42d4e-99bb-4fc2-ad1c-8efaa1d2d995)

</details>

effigies commented on mkdocs.yml:
```diff
@@ -132,7 +136,10 @@ plugins:
         "01-introduction.md": "introduction.md"
         "02-common-principles.md": "common-principles.md"
-        "03-modality-agnostic-files.md": "modality-agnostic-files.md"
+        "03-modality-agnostic-files/01-dataset-description.md": "modality-agnostic-files/dataset-description.md"
+        "03-modality-agnostic-files/02-data-summary-files.md": "modality-agnostic-files/data-summary-files.md"
+        "03-modality-agnostic-files/03-phenotypic-and-assessment-data.md": "modality-agnostic-files/phenotypic-and-assessment-data.md"
+        "03-modality-agnostic-files/04-code.md": "modality-agnostic-files/code.md"
```
This breaks old links that point to `03-modality-agnostic-files.html`, while these new files never existed, so don't need redirects. I think we just want:

```suggestion
        "03-modality-agnostic-files.md": "modality-agnostic-files/dataset-description.md"
```

Once https://github.com/mkdocs/mkdocs-redirects/pull/73 is released, we could redirect the headings to the appropriate files.

yarikoptic commented on mkdocs.yml:
```diff
@@ -132,7 +136,7 @@ plugins:
         "01-introduction.md": "introduction.md"
         "02-common-principles.md": "common-principles.md"
-        "03-modality-agnostic-files.md": "modality-agnostic-files.md"
+        "03-modality-agnostic-files.md": "modality-agnostic-files/dataset-description.md"
```
I think we need to keep old and add the unnumbered one, so

```suggestion
        "03-modality-agnostic-files.md": "modality-agnostic-files/dataset-description.md"
        "modality-agnostic-files.md": "modality-agnostic-files/dataset-description.md"
```

because currently we have 

https://bids-specification.readthedocs.io/en/stable/modality-agnostic-files.html

so it is the one to redirect too.  And indeed then to complement all those section redirects appropriately after 
- https://github.com/mkdocs/mkdocs-redirects/pull/73
is released

cmaumet commented:
Hey @yarikoptic @effigies are we good to merge with your two approvals or should we wait for more input? Sorry just not sure about the process here... Thanks!

effigies commented:
I think we can merge. Been over a week since significant changes, just touch-ups since then.

yarikoptic commented:
but ideally we are still to add more redirects whenever someone (might be us) brings that https://github.com/mkdocs/mkdocs-redirects/pull/73 over the finish line
